Sprint_6

Andrushka888 opened the pull request:
К проекту подключены Jacoco, Mockito и JUnit;
Скорректирован класс Lion - убрана зависимость от Feline; Написаны тесты для Feline, Cat и Lion. При написании, для некоторых тестов были использованы Mock и Параметризация; Так же был сформировано отчет Jacoco, в котором выполнено покрытие более 70%.

Andrushka888 committed:
К проекту подключены Jacoco, Mockito и JUnit;
Скорректирован класс Lion - убрана зависимость от Feline;
Написаны тесты для Feline, Cat и Lion. При написании, для некоторых тестов были использованы Mock и Параметризация;
Так же был сформировано отчет Jacoco, в котором выполнено покрытие более 70%.

mkouznetsov-git commented on src/main/java/com/example/Lion.java:
```diff
@@ -16,7 +16,11 @@ public Lion(String sex) throws Exception {
-    Feline feline = new Feline();
+    private Feline feline;
+
+    public Lion(Feline feline) { // Принцип инъекции зависимостей. Теперь Lion не зависит от Feline
```
У тебя получилось два конструктора, один делает льва совсем без Feline, другой делает льва без пола. Это некорректно.

mkouznetsov-git commented on src/test/java/TestLion.java:
```diff
@@ -0,0 +1,83 @@
+
+
+    @Test
+    public void TestLionHaveMane() throws Exception {
```
Этот тест не проходит. Тебе надо разобраться, в чём дело и поправить.

mkouznetsov-git commented on src/test/java/TestCat.java:
```diff
@@ -0,0 +1,45 @@
+
+        Cat cat = new Cat(predator);
+
+        Mockito.when(predator.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
```
Зачем использовать здесь мок с возвращаемыми значениями, которые и без мока возвращаются такие же?

Andrushka888 committed:
Был скорректирован Lion, два конструктора были объединены в один;
Поправлены название теста и наименования переменных в тесте "doesHaveMane()" на более информативные.

mkouznetsov-git commented on src/test/java/TestLion.java:
```diff
@@ -0,0 +1,84 @@
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
```
4 из 9 тестов падают. Надо поправить логику проверки. Сам тестируемый код написан корректно.

mkouznetsov-git replied:
У тебя логика проверки становится уже сложнее, чем сам проверяемый код. )
У тебя нет проверки, что если sex=female, то hasMane() возвращает false.
Подсказка: добавь в параметры ожидаемый результат.

Andrushka888 committed:
В класс TestLion добавлена обработка исключений через конструктор try и catch

Andrushka888 committed:
В класс TestLion добавлен ожидаемый результат в параметризацию
